src/main/java/org/cloudfoundry/promregator/endpoint/MetricsEndpoint.java

```diff
@@ -63,6 +67,10 @@ protected boolean isLabelEnrichmentSuppressable() {
 	 */
 	@Override
 	protected List<MetricsFetcher> createMetricsFetchers(List<Instance> instanceList) {
+		if(maxAllowedForSingleEndpoint != -1 && instanceList.size() > maxAllowedForSingleEndpoint) {
+			throw new RuntimeException("promregator.single.endpoint.max is configured to only allow "+maxAllowedForSingleEndpoint+" instances but there were "+ instanceList.size());
```

Comment:
I am asking myself right now whether it wouldn't be better to send a proper error message with a good HTTP status code here instead.

Reply:
It would be. A `503` would probably be a good idea. If you think this feature is worth having.